xarray/backends/h5netcdf_.py

```diff
@@ -69,8 +69,40 @@ class H5NetCDFStore(WritableCFDataStore):
     """Store for reading and writing data via h5netcdf
     """
 
-    def __init__(
-        self,
+    __slots__ = (
+        "autoclose",
+        "format",
+        "is_remote",
+        "lock",
+        "_filename",
+        "_group",
+        "_manager",
+        "_mode",
+    )
+
+    def __init__(self, manager, group=None, mode=None, lock=HDF5_LOCK, autoclose=False):
+
+        import h5netcdf
+
+        if isinstance(manager, (h5netcdf.File, h5netcdf.Group)):
+            if group is None:
+                root, group = manager._root, manager.name
+            else:
+                root = manager
+            manager = DummyFileManager(root)
+
+        self._manager = manager
+        self._group = group
+        self._mode = mode
+        self.format = None
+        self._filename = self.ds._root.filename
```

Comment:
Likewise, we don't want to rely on a private API here

Reply:
Unfortunately `h5netcdf.Group` doesn't seem to have a `filename` property like `netCDF4.Dataset`. So there is currently no direct means of extracting the filename from `h5netcdf.Group` using public API. Trying to to facilitate `find_root_and_group` now.